Formly redesign

Allcharles opened the pull request:
# Formly Redesign (Do Not Merge)

Re-implementation/redesign of formly wrapper component and form template. The intention is for this PR to change the formly implementation to a sustainable solution. Other PRs will come later (and merge into this) to update the rest of the pages and unit tests.

## Changes

- Created `formTemplate`
- Changed `<app-form>` inputs to accept `model` and `fields`

## Issues

- Not completed, will create additional PR's to merge into this before accepting the final changes

## Closes 

Closes #157

Allcharles committed:
Create formTemplate class

atruskie commented on src/app/component/projects/pages/edit/edit.component.ts:
```diff
@@ -35,75 +32,58 @@ import { fields } from "./edit.json";
+          errMsg = err.message;
+        }
+        return errMsg;
+      }
```
these both look fairly standard - refactor opportunity?

atruskie commented on src/app/component/projects/pages/edit/edit.component.ts:
```diff
@@ -35,75 +32,58 @@ import { fields } from "./edit.json";
+      notifications,
+      route,
+      router,
+      ["project"],
```
string token here is concerning

atruskie commented on src/app/component/shared/form/form.component.ts:
```diff
@@ -26,69 +23,38 @@ import { ApiResponse } from "src/app/services/baw-api/baw-api.service";
-    | "btn-info" = "btn-success";
   /**
-   * Deprecated
+   * DEPRECATED
```
the diff is a little hard to read - is this entire component deprecated?

Allcharles replied:
No, just `@Input() schema: any`. I haven't removed it so that I can still compile the app while i make changes to the new inputs: `model`, and `fields`

atruskie commented on src/app/helpers/formTemplate/formTemplate.ts:
```diff
@@ -0,0 +1,118 @@
+    protected notifications: ToastrService,
+    protected route: ActivatedRoute,
+    protected router: Router,
+    private modelKeys: string[],
```
so this relates to my previous string focused comment. 

I wonder, can we attach to models (maybe to AbsrtactModel) a method that returns the name of the model?
Then all we'd need to pass here is any model or model constructor ...

Allcharles replied:
So i already thought about that, and the `kind` key will tell us what type of `AbstractModel` we are dealing with. However i couldnt figure out the next step of typing the variable.

atruskie commented on src/app/helpers/formTemplate/formTemplate.ts:
```diff
@@ -0,0 +1,118 @@
+    protected router: Router,
+    private modelKeys: string[],
+    private successMsg: (model: Partial<M>) => string,
+    private errorMsg: (err: ApiErrorDetails) => string = err => err.message
```
these could probably have defaults set

atruskie reviewed (changes requested):
This looks much cleaner than #153 - well done.

A little bit more work to clean-up those stringly-typed data keys and we'll be set

Allcharles committed:
Cleanup rebase mistakes

Allcharles committed:
Form Reintegration (#165)

- Reintegrated forms
- Enabled disabling of FormCheck gaurd
- Cleanup user-badge unit tests
- Added `FormTemplate` unit tests
- Updated AccountService
- Disabled formly timezone
